packages/roosterjs-editor-plugins/lib/HyperLink/HyperLink.ts

```diff
@@ -166,20 +172,20 @@ export default class HyperLink implements EditorPlugin {
      * The reason this is put in a try-catch is that
      * it has been seen that accessing href may throw an exception, in particular on IE/Edge
      */
-    private tryGetHref(element: Element): string {
-        let href: string = null;
+    private tryGetHref(anchor: HTMLAnchorElement): string {
         try {
-            do {
-                if (element.tagName == 'A') {
-                    href = (<HTMLAnchorElement>element).href;
-                    break;
-                }
-                element = element.parentElement;
-            } while (this.editor.contains(element));
-        } catch (error) {
-            // Not do anything for the moment
-        }
+            return anchor ? anchor.href : null;
+        } catch {}
+    }
+
+    private tryGetAnchor(element: Element): HTMLAnchorElement {
+        do {
+            if (element.tagName == 'A') {
+                return <HTMLAnchorElement>element;
+            }
+            element = element.parentElement;
+        } while (this.editor.contains(element));
 
-        return href;
+        return null;
     }
```

Comment:
You can use editor.getElementAtCursor('A', element); It does the same thing with this function